src/authentication/AuthenticatedRoute.tsx

```diff
@@ -19,30 +19,37 @@ const AuthenticatedRouteRedirecter: FC<AuthenticatedRouteRedirecterProps> = ({
     return <Redirect to="/login" />;
   }
 
-  if (requiredPermission && !hasPermission(requiredPermission)) {
+  if (requiredPermissions && !hasPermissions(requiredPermissions, hasPermission)) {
     return <NotFoundPage />;
   }
 
   return <>{children}</>;
 };
 
 interface AuthenticatedRouteProps extends RouteProps {
-  requiredPermission?: string;
+  requiredPermissions?: string[];
 }
 
 export const AuthenticatedRoute: FC<AuthenticatedRouteProps> = ({
-  requiredPermission,
+  requiredPermissions,
   children,
   ...rest
 }) => {
   return (
     <Route
       {...rest}
       render={() => (
-        <AuthenticatedRouteRedirecter requiredPermission={requiredPermission}>
+        <AuthenticatedRouteRedirecter requiredPermissions={requiredPermissions}>
           {children}
         </AuthenticatedRouteRedirecter>
       )}
     />
   );
 };
+
+function hasPermissions(
+  permissions: string[],
+  hasPermission: (permission: string) => boolean
+): boolean {
+  return permissions.map(hasPermission).every((value) => value === true);
```

Comment:
```suggestion
  return permissions.map(hasPermission).every(Boolean);
```

Reply:
Also, worth thinking about replacing the usage of `hasPermissions` entirely with `permissions.every(hasPermission)` I think it's easier to understand

Reply:
I thought the short version is a bit misleading, as the permissions don't really have permissions, but making it longer didn't really make it clear. Will make changes.

Reply:
Aliased `hasPermission` as `userHasPermissions` and inlined:
```js
!requiredPermissions.every(userHasPermission)
```